### PR TITLE
Release/7.3.0

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

File: test/test-pages/gcba/sdk-controller.ts [hunk 1]
```diff
@@ -0,0 +1,307 @@
+/* global MediaStream */
+import GenesysCloudWebrtSdk, { ConversationUpdate, IExtendedMediaSession, ISdkConfig, ISdkConversationUpdateEvent, IStoredConversationState, SdkError, SessionTypes } from '../../../src';
+import { getInputValue, writeToLog } from './utils';
+
+let webrtcSdk: GenesysCloudWebrtSdk;
+let pendingSessions = [];
+
+let conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:8]: Codacy has a fix for the issue: ['conversationUpdatesToRender' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
const conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
```

File: test/test-pages/gcba/platform-client.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    redirectUri += window.location.pathname;
+    console.debug('Signing out with redirect of: ' + redirectUri);
+
+    client.logout(redirectUri);
+  }
+
+  function authenticate () {
+    const environment = getInputValue('environment');
+    const clientId = (window as any).environments[environment].clientId;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:37]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    const clientId = (window as unknown).environments[environment].clientId;
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 3]
```diff
@@ -153,3 +150,91 @@ describe('updateOutgoingMedia', () => {
...
+
+    jest.spyOn(utils, 'requestApi').mockResolvedValue(null);
+
+    await handler['updateScreenRecordingMetadatas'](session as any, metas as any);
+
+    expect(metas[0].trackId).toBe('3');
+    expect((metas[0] as any)._trackId).toBe('trackId123');
+
+    expect(metas[1].trackId).toBe('7');
+    expect((metas[1] as any)._trackId).toBe('trackId456');
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:200]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    expect((metas[1] as unknown)._trackId).toBe('trackId456');
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  options.environment = environmentData.uri;
+  options.logLevel = 'info';
+  // for sumo debugging
+  // options.optOutOfTelemetry = true;
+
+  const SDK = GenesysCloudWebrtSdk;
+  (window as any).SDK = SDK;
+
+  webrtcSdk = new SDK(options);
+  (window as any).webrtcSdk = webrtcSdk;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:33]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).webrtcSdk = webrtcSdk;
```

File: test/test-pages/gcba/platform-client.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    initClientEnvironment(environment);
+
+    client.loginImplicitGrant(clientId, window.location.href)
+      .then(() => {
+        const authInfo = JSON.parse(window.localStorage.getItem(`${persitentName}_auth_data`));
+        (client as any).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:43]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
        (client as unknown).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+  (window as any).sdk = webrtcSdk;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:34]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).sdk = webrtcSdk;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+function ready () {
+  if (!webrtcSdk._personDetails) {
+    webrtcSdk.startScreenShare();
+  }
+  writeToLog('webrtcSDK ready event emitted');
+}
+
+// pendingSession - {id, address, conversationId, autoAnswer}
+function pendingSession (options) {
+  let output = `${_getLogHeader('pendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:102]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('pendingSession')}
```

File: test/test-pages/gcba/platform-client.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+        (window as any).conversationsAPI = new (window as any).platformClientConversationsApi();
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
        (window as unknown).conversationsAPI = new (window as any).platformClientConversationsApi();
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+    autoAcceptPendingScreenRecordingRequests: true
+  }
+
+  options.environment = environmentData.uri;
+  options.logLevel = 'info';
+  // for sumo debugging
+  // options.optOutOfTelemetry = true;
+
+  const SDK = GenesysCloudWebrtSdk;
+  (window as any).SDK = SDK;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:30]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).SDK = SDK;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+
+function error (error: SdkError) {
+  let output = `${_getLogHeader('error')}
+    error: ${error.message}\n  details: ${error.details}`;
+
+  writeToLog(output);
+}
+
+function sessionEnded (session, reason) {
+  let output = `${_getLogHeader('terminated')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:291]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('terminated')}
```

File: test/unit/media/headset.test.ts [hunk 11]
```diff
@@ -167,6 +167,31 @@ describe('SdkHeadset', () => {
...
+      const conversationId = 'myConvoId';
+      sdk.emit('sessionEnded', { conversationId } as any, null);
+
+      expect(sdkHeadset.endCurrentCall).toHaveBeenCalledWith(conversationId);
+    });
+
+    it('should set ringing if not autoanswer', () => {
+      sdkHeadset.setRinging = jest.fn();
+
+      sdk.emit('pendingSession', { autoAnswer: false, conversationId: '15215asdf'} as any);
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:184]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
      sdk.emit('pendingSession', { autoAnswer: false, conversationId: '15215asdf'} as unknown);
```

File: test/test-pages/gcba/platform-client.ts [hunk 12]
```diff
@@ -0,0 +1,170 @@
+import { getCurrentUrlParams, getInputValue } from './utils';
+const client = (window as any).platformClient.ApiClient.instance;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:2]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
const client = (window as unknown).platformClient.ApiClient.instance;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+function rejectPendingSession ({ conversationId }) {
+  webrtcSdk.rejectPendingSession({ conversationId });
+}
+
+function endSession ({ conversationId }) {
+  webrtcSdk.endSession({ conversationId });
+}
+
+function exposeGlobalFunctions () {
+  (window as any).acceptPendingSession = acceptPendingSession;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:70]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).acceptPendingSession = acceptPendingSession;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+  (window as any).rejectPendingSession = rejectPendingSession;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:71]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).rejectPendingSession = rejectPendingSession;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 15]
```diff
@@ -0,0 +1,307 @@
...
+  (window as any).endSession = endSession;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:72]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).endSession = endSession;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 16]
```diff
@@ -0,0 +1,307 @@
...
+  ];
+
+  webrtcSdk.acceptSession({ conversationId: session.conversationId, sessionType: session.sessionType, mediaStream: screenStream, screenRecordingMetadatas });
+}
+
+function trace (level, message, details) {
+  let output = `${_getLogHeader('trace')}\n`;
+  output += `  level: ${level}\n  message: ${message}\n  details: ${details}`;
+
+  const logTraces = (document.getElementById('log-traces-check') as any).checked;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:277]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const logTraces = (document.getElementById('log-traces-check') as unknown).checked;
```

**Review by zservies — APPROVED** [review]

File: test/test-pages/gcba/platform-client.ts (hunk 12 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:2]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
const client = (window as unknown).platformClient.ApiClient.instance;
```

File: test/unit/client-private.test.ts [hunk 17]
```diff
@@ -2,15 +2,59 @@ import { GenesysCloudWebrtcSdk } from '../../src/client';
...
+      chat: {
+        jabberId: 'myJid'
+      }
+    };
+
+    let connectedCb: () => void;
+    const connectSpy = jest.fn().mockImplementation(() => {
+      connectedCb();
+    });
+    (StreamingClient as any).mockReturnValue({
```
Comment (davidmorton-gen) [inline test/unit/client-private.test.ts:40]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    (StreamingClient as unknown).mockReturnValue({
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 18]
```diff
@@ -0,0 +1,307 @@
...
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+async function sessionStarted (session: IExtendedMediaSession) {
+  let output = `${_getLogHeader('sessionStarted')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:244]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('sessionStarted')}
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 19]
```diff
@@ -153,3 +150,91 @@ describe('updateOutgoingMedia', () => {
...
+
+    const session = {
+      pc: {
+        getTransceivers: jest.fn().mockReturnValue(transceivers)
+      }
+    };
+
+    jest.spyOn(utils, 'requestApi').mockResolvedValue(null);
+
+    await handler['updateScreenRecordingMetadatas'](session as any, metas as any);
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:194]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    await handler['updateScreenRecordingMetadatas'](session as unknown, metas as any);
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 20]
```diff
@@ -0,0 +1,307 @@
...
+    <td class="${isSessionActive ? 'text-success' : 'text-danger'}">
+      ${session.state}
+    </td>
+  </tr>`
+  });
+  tableBody.innerHTML = html;
+}
+
+function cancelPendingSession (params) {
+  let output = `${_getLogHeader('cancelPendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:224]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('cancelPendingSession')}
```

File: test/test-pages/gcba/platform-client.ts [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  async function fetchNewJwt () {
+    const accessToken = manualAuthInput.value;
+
+    if (!accessToken) {
+      throw new Error('Must have an access token to');
+    }
+
+    const url = `https://api.${(window as any).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:120]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    const url = `https://api.${(window as unknown).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```

File: test/test-pages/gcba/platform-client.ts [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+      });
+  }
+
+  function authenticateFromToken (token) {
+    const environment = getInputValue('environment');
+    initClientEnvironment(environment);
+
+    window.localStorage.setItem(`${persitentName}_auth_data`, JSON.stringify({ accessToken: token }));
+    client.authentications['PureCloud OAuth'].accessToken = token;
+    (window as any).conversationsAPI = new (window as any).platformClient.ConversationsApi();
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:59]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    (window as unknown).conversationsAPI = new (window as any).platformClient.ConversationsApi();
```

File: test/unit/media/headset.test.ts [hunk 23]
```diff
@@ -167,6 +167,31 @@ describe('SdkHeadset', () => {
...
+      expect(sdkHeadset.setRinging).toHaveBeenCalled();
+    });
+
+    it('should not set ringing if autoanswer', () => {
+      sdkHeadset.setRinging = jest.fn();
+
+      sdk.emit('pendingSession', { autoAnswer: true, conversationId: '15215asdf'} as any);
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:191]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
      sdk.emit('pendingSession', { autoAnswer: true, conversationId: '15215asdf'} as unknown);
```

File: test/test-pages/gcba/platform-client.ts [hunk 24]
```diff
@@ -0,0 +1,170 @@
...
+    client.setEnvironment((window as any).environments[environment].uri);
+  }
+
+  function setAuthTextVisible (visible) {
+    const style = visible ? 'visible' : 'hidden';
+    document.getElementById('auth-text').style.visibility = style;
+  }
+
+  function signout () {
+    const environment: any = getInputValue('environment');
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:17]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    const environment: unknown = getInputValue('environment');
```

File: src/sessions/screen-recording-session-handler.ts [hunk 25]
```diff
@@ -73,15 +73,37 @@ export default class ScreenRecordingSessionHandler extends BaseSessionHandler {
   }
 
   private updateScreenRecordingMetadatas (session: ScreenRecordingMediaSession, metadatas: ScreenRecordingMetadata[]) {
+    metadatas.forEach((meta) => {
+      // adding any here because I don't want to add this to the public interface
+      (meta as any)._trackId = meta.trackId;
```
Comment (davidmorton-gen) [inline src/sessions/screen-recording-session-handler.ts:78]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
      (meta as unknown)._trackId = meta.trackId;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 26]
```diff
@@ -0,0 +1,307 @@
...
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+function handledPendingSession (params) {
+  let output = `${_getLogHeader('handledPendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:234]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('handledPendingSession')}
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 13 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:70]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).acceptPendingSession = acceptPendingSession;
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 14 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:71]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).rejectPendingSession = rejectPendingSession;
```

File: test/unit/media/headset.test.ts [hunk 27]
```diff
@@ -167,6 +167,31 @@ describe('SdkHeadset', () => {
       expect(setHoldSpy).toHaveBeenCalledWith('123', false);
     });
   });
+
+  describe('listenForSessionEvents', () => {
+    it('should end currentCall on sessionEnded', () => {
+      sdkHeadset.endCurrentCall = jest.fn();
+
+      const conversationId = 'myConvoId';
+      sdk.emit('sessionEnded', { conversationId } as any, null);
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:176]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
      sdk.emit('sessionEnded', { conversationId } as unknown, null);
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 4 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:33]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).webrtcSdk = webrtcSdk;
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 6 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:34]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  (window as unknown).sdk = webrtcSdk;
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 28]
```diff
@@ -153,3 +150,91 @@ describe('updateOutgoingMedia', () => {
...
+
+    expect(metas[0].trackId).toBe('3');
+    expect((metas[0] as any)._trackId).toBe('trackId123');
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:197]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
    expect((metas[0] as unknown)._trackId).toBe('trackId123');
```

File: test/unit/media/media.test.ts [hunk 29]
```diff
@@ -1482,6 +1482,18 @@ describe('SdkMedia', () => {
       expect(sdk.sessionManager.validateOutgoingMediaTracks).toHaveBeenCalled();
       expect(sdk.logger.debug).toHaveBeenCalled();
     });
+
+    it('should not call sessionManager if it has not been initialized yet', async () => {
+      const enumerateDevicesSpy = jest.spyOn(sdkMedia, 'enumerateDevices').mockResolvedValue([]);
+      const miniSdk = { logger: { debug: jest.fn() } };
+      sdkMedia['sdk'] = miniSdk as any;
```
Comment (davidmorton-gen) [inline test/unit/media/media.test.ts:1489]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
      sdkMedia['sdk'] = miniSdk as unknown;
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 1 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:8]: Codacy has a fix for the issue: ['conversationUpdatesToRender' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
const conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
```

**Pull request merged**